float8_experimental/float8_linear_utils.py

```diff
@@ -117,8 +117,23 @@ def swap_linear_with_float8_linear(
             swap_linear_with_float8_linear(child, module, emulate)
 
 
+def get_float8_layers(model: torch.nn.Module, fp8_classes=None):
+    if fp8_classes is None:
+        # Lazy import to avoid circular dependency
+        from float8_experimental.float8_linear import Float8Linear
+
+        fp8_classes = Float8Linear
+
+    # Get all fp8 layers and tensors
+    fp8_layers = [
+        child for name, child in model.named_modules() if isinstance(child, fp8_classes)
+    ]
+
+    return fp8_layers
+
+
 def sync_float8_amax_and_scale_history(
-    model: torch.nn.Module, fp8_classes=None
+    model: torch.nn.Module, fp8_classes=None, fp8_layers=None, combine_reduction=False
```

Comment:
thoughts on splitting these changes into a separate diff? something like
1. diff 1: add microbenchmark for FSDP
2. diffs 2..n: improvements to sync function, with before+after benchmark numbers

Reply:
Thank you for the comments. Let me split into small PRs.